provider/common.go

```diff
@@ -348,12 +348,7 @@ func marshalBasicTypesByTag(tag string, in map[string]interface{}, out any) (dia
 		if value, ok := in[_tag]; ok && value != nil {
 			switch field.Type().Name() {
 			case "StringValue":
-				val := fmt.Sprintf("%v", value)
-				if val == "" {
-					field.Set(reflect.ValueOf(types.StringNull()))
-				} else {
-					field.Set(reflect.ValueOf(types.StringValue(val)))
-				}
```

Comment:
Problem: null <-> “” string inconsistency. I always set in state ValueString(“”), on Import & Read & Update.
if Attribute required -> framework dumps empty string as empty string (“”)
If Attribute optional -> framework dumps empty string as null, but then on Read it decides to read it as empty string, generates a plan, and doesn’t display change  (it looks like nothing changed but resource is marked to be changed)
If Attributes optional+computed it dumps empty string as null and read empty string as null


// All optional + computed fields are set to null /default values in plan modifier, in case plan is Unknown(), this leads to clean an expected plans